src/odemis/acq/stream/_static.py

```diff
@@ -607,19 +609,33 @@ def __init__(self, name, image, *args, **kwargs):
                                                        unit=unit_bw,
                                                        setter=self._setWavelength)
 
-        # Is there time data?
+        # Is there time or theta data?
         if image.shape[1] > 1:
             # cached list of timestamps for each position in the time dimension
-            tl, unit_t = spectrum.get_time_range(image)
-            self._tl_px_values = numpy.array(tl, copy=False)  # Force it to be a numpy array
-            min_t, max_t = self._tl_px_values[0], self._tl_px_values[-1]
-
-            # Allow the select the time as any value within the range, and the
-            # setter will automatically "snap" it to the closest existing timestamp
-            self.selected_time = model.FloatContinuous(self._tl_px_values[0],
-                                                   range=(min_t, max_t),
-                                                   unit=unit_t,
-                                                   setter=self._setTime)
+            if model.MD_TIME_LIST in image.metadata:
+                tl, unit_t = spectrum.get_time_range(image)
+                self._tl_px_values = numpy.array(tl, copy=False)  # Force it to be a numpy array
+                min_t, max_t = self._tl_px_values[0], self._tl_px_values[-1]
+
+                # Allow the select the time as any value within the range, and the
+                # setter will automatically "snap" it to the closest existing timestamp
+                self.selected_time = model.FloatContinuous(self._tl_px_values[0],
+                                                           range=(min_t, max_t),
+                                                           unit=unit_t,
+                                                           setter=self._setTime)
+            else:  # MD_THETA_LIST is present
+                # cached list of theta stamps for each position in the theta dimension
+                theta_list, unit_theta = spectrum.get_angle_range(image)
+                # Converts radians to degrees and removes NaN values from the theta list
+                self._thetal_px_values = [math.degrees(theta) for theta in theta_list]
```

Comment:
I think it'd be simpler to not convert to degrees (yet). Only do the conversion at display in the GUI (if the unit is "rad").
Also, here you are not removing the NaNs anymore. It seems to work fine for min/max (they discard NaN).
However for searchsorted() it might not work. Now that the calibrated image has no NaN, so probably you need to remove NaN also here.